### PR TITLE
test(ci): add SIGSEGV handler integration tests and musl CI job

### PR DESCRIPTION
**Pull request by jackchuma**
## Summary

- Add integration tests for the SIGSEGV signal handler that verify segfaults (null dereference) and stack overflows produce meaningful diagnostic output with backtraces and cycle detection.
- Tests use a self-re-invocation pattern to safely trigger faults in isolated child processes.
- Add a dedicated `test-musl-sigsegv` CI job that runs the tests under `x86_64-unknown-linux-musl` to exercise the alternate signal stack in a static-linking environment.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>✅ Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
✅
<code>1/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: crates/utilities/cli/tests/sigsegv_test.rs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+fn segfault_produces_backtrace() {
+    dispatch();
+
+    let output = run_child("segfault_produces_backtrace", "null_deref");
+    let stderr = String::from_utf8_lossy(&output.stderr);
+
+    assert_killed_by_sigsegv(&output, &stderr);
+
+    assert!(
+        stderr.contains("0x"),
```
Comment (leopoldjoy) [inline crates/utilities/cli/tests/sigsegv_test.rs:133]: Nit: any hex address in any output would satisfy this (including error messages unrelated to the backtrace). Consider asserting on a line count or a more specific pattern, as it could false-pass.

**Review by leopoldjoy — APPROVED** [review]
